lib/utils/platform/iot_threads_afr.c

```diff
@@ -0,0 +1,350 @@
+     */
+    #define IotThreads_Malloc    malloc
+#endif
+#ifndef IotThreads_Free
+    #include <stdlib.h>
+
+    /**
+     * @brief Free memory. This function should have the same signature as
+     * [free](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html).
+     */
+    #define IotThreads_Free    free
+#endif
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @defgroup Mutex types.
+ */
+/**@{ */
+#ifndef IOT_THREAD_MUTEX_NORMAL
+    #define IOT_THREAD_MUTEX_NORMAL        0                        /**< Non-robust, deadlock on relock, does not remember owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_ERRORCHECK
+    #define IOT_THREAD_MUTEX_ERRORCHECK    1                        /**< Non-robust, error on relock,  remembers owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_RECURSIVE
+    #define IOT_THREAD_MUTEX_RECURSIVE     2                        /**< Non-robust, recursive relock, remembers owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_DEFAULT
+    #define IOT_THREAD_MUTEX_DEFAULT       IOT_THREAD_MUTEX_NORMAL  /**< IOT_THREAD_MUTEX_NORMAL (default). */
+#endif
+/**@} */
+
+/*-----------------------------------------------------------*/
+
+static void _threadRoutineWrapper( void * pArgument )
+{
+    _threadInfo_t * pThreadInfo = ( _threadInfo_t * ) pArgument;
+
+    /* Run the thread routine. */
+    pThreadInfo->threadRoutine( pThreadInfo->pArgument );
+    IotThreads_Free( pThreadInfo );
+
+    vTaskDelete( NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+bool Iot_CreateDetachedThread(  IotThreadRoutine_t threadRoutine,
+                                void * pArgument,
+                                int32_t priority,
+                                size_t stackSize )
+{
+    IotLogDebug( "Creating new thread." );
+    int status = 0;
+    _threadInfo_t * pThreadInfo = IotThreads_Malloc(sizeof(_threadInfo_t));
+
+    if ( pThreadInfo == NULL)
+    {
+        IotLogDebug( "Unable to allocate memory for threadRoutine %p.", threadRoutine );
+        status = 1;
+    }
+
+    /* Create the FreeRTOS task that will run the thread. */
+    if ( status == 0 )
+    {
+        pThreadInfo->threadRoutine = threadRoutine;
+        pThreadInfo->pArgument = pArgument;
+        
+        /* update supplied if we require the default stack size */
+        if ( stackSize == IOT_THREAD_DEFAULT_STACK_SIZE)
+        {
+            stackSize = 3072;
+        }
+
+        /* update supplied if we require the default thread priority */
+        if ( priority == IOT_THREAD_DEFAULT_PRIORITY )
+        {
+            priority = 5;
+        }
+
+        if( xTaskCreate( _threadRoutineWrapper,
+                        "iot_thread",
+                        stackSize,
+                        pThreadInfo,
+                        priority,
+                        NULL ) != pdPASS )
+        {
+            /* Task creation failed. */
+            IotLogWarn( "Failed to create thread.\r\n" );
```

Comment:
Maybe just one call?

Reply:
The second one was test code I intended to remove and missed! Do you think we should keep it?